controllers/rcj_soccer_referee_supervisor/recorder/recorder.py

```diff
@@ -44,19 +44,22 @@ def get_resolution(self):
 
         return res_table[self.resolution]
 
-    def start_recording(self):
+    def start_recording(self, formats):
         width, height = self.get_resolution()
         filename = self.create_title()
 
         # API details for movieStartRecording
         # https://www.cyberbotics.com/doc/reference/supervisor?tab-language=python#wb_supervisor_movie_start_recording
-        self.supervisor.movieStartRecording(filename,
-                                            width,
-                                            height,
-                                            quality=100,
-                                            codec=0,
-                                            acceleration=self.fastforward_rate,
-                                            caption=False)
+        if 'mp4' in formats:
```

Comment:
Let's create a class `class RecordingFormats(Enum)` in conts.py for this, and add `mp4` and `x3d` there.